src/main/java/com/puppycrawl/tools/checkstyle/SuppressionsStringPrinter.java

```diff
@@ -0,0 +1,152 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2018 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Pattern;
+
+import antlr.RecognitionException;
+import antlr.TokenStreamException;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FileContents;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+import com.puppycrawl.tools.checkstyle.xpath.XpathQueryGenerator;
+
+/**
+ * Class for generating and printing suppressions.
+ * @author Timur Tibeyev.
+ */
+public final class SuppressionsStringPrinter {
```

Comment:
The class prints xpath suppressions. The name should contain this fact.

FYI:
Checkstyle has classes which 'prints' the content of AST tree or its parts. Truth be told, these classes designed incorrectly, as for me. They do not print objects, they construct strings. From my point of view, the correct pattern for printers is to print something to the stream (OutputStream, File, etc). The printer should know nothing about the stream which receives the information except its interface. So the printer is similar to the formatter. It prepares and formats the information for the output stream. For more information see http://www.yegor256.com/2016/04/05/printers-instead-of-getters.html. It is not the case of this PR, lets do not break the current Checkstyle printers model, it is just something for you to think about.

Reply:
done